### PR TITLE
Support native union return typing

### LINKED ISSUE #462: PHP 8 Union type fails while union type in doc block works
I get this error when using the output type:

```php
#[Field]
public function origin(Foo $foo): Bar|Baz
{
  return $foo->origin;
}
```

```
In TypeHandler.php line 367:

  assert($type instanceof ReflectionNamedType)
```

The output types for Bar and Baz are already defined.

Adding the union type to the docblock works:

```php
/**
* @return Bar|Baz
*/
#[Field]
public function origin(Foo $foo)
{
  return $foo->origin;
}
```

I assume this is a missing php 8 features

### PR DESCRIPTION
**Pull request by aszenz**
Fixes #462

### CODE REVIEW

File: tests/Mappers/Parameters/TypeMapperTest.php [hunk 1]
```diff
@@ -77,6 +101,10 @@ private function dummy() {
 
     }
 
+    private function objectUnion(): TestObject|TestObject2 {
```
Comment (aszenz) [inline tests/Mappers/Parameters/TypeMapperTest.php:104]: I also tried `null|TestObject|TestObject2` but that failed, not sure if the library needs to support that

**Comment by aszenz** [comment]
@oojacoboo Any idea how to fix the test for php 7.4

**Comment by oojacoboo** [comment]
@aszenz PHP didn't support union types until 8.0.  The parser for 7.4 doesn't understand the syntax.  Makes sure the tests are only running for 8.0.  You can look at how `tests/Fixtures81` is implemented.  If necessary, create a new `Fixtures80` directory.

Tests are run through `.github/workflows/continuous_integration.yml`.

**Comment by aszenz** [comment]
> @aszenz PHP didn't support union types until 8.0. The parser for 7.4 doesn't understand the syntax. Make sure the tests are only running for 8.0. You can look at how `tests/Fixtures81` is implemented. If necessary, create a new `Fixtures80` directory.
> 
> Tests are run through `.github/workflows/continuous_integration.yml`

Done :+1:

File: src/Mappers/Parameters/TypeHandler.php [hunk 2]
```diff
@@ -408,17 +410,36 @@ private function appendTypes(Type $type, ?Type $docBlockType): Type
...
+                $phpdocType = new Nullable($phpdocType);
+            }
 
-        return $phpdocType;
+            return $phpdocType;
+        }
+        return new Compound(
+            array_map(
+                function ($namedType) use ($reflectionClass): Type {
+                    \assert($namedType instanceof ReflectionNamedType);
```
Comment (aszenz) [inline src/Mappers/Parameters/TypeHandler.php:429]: had to add this assert to satisfy phpstan for php7.4

**Comment by oojacoboo** [comment]
@aszenz awesome thanks and for the PR!

**Pull request merged**